lib/compat/wordpress-7.0/blocks.php

```diff
@@ -121,3 +121,63 @@ function gutenberg_resolve_pattern_blocks( $blocks ) {
 		return $blocks;
 	}
 }
+
+/**
+ * Adds update `taxQuery` args to handle exclusion of taxonomy terms.
+ *
+ * @see 'query_loop_block_query_vars'
+ *
+ * @param array    $query The query vars.
+ * @param WP_Block $block Block instance.
+ * @return array   The filtered query vars.
+ */
+function gutenberg_update_tax_query_of_query_loop_block( $query, $block ) {
+	if ( empty( $block->context['query']['taxQuery'] ) || ! is_array( $block->context['query']['taxQuery'] ) ) {
+		return $query;
+	}
+
+	// If there are keys other than include/exclude, it's the old
+	// format and has been handled already.
+	if ( ! empty( array_diff( array_keys( $block->context['query']['taxQuery'] ), array( 'include', 'exclude' ) ) ) ) {
+		return $query;
+	}
+
+	// Build with the new structure.
+	$tax_query_input = $block->context['query']['taxQuery'];
+
+	// Helper function to build tax_query conditions from taxonomy terms.
+	$build_conditions = static function ( $terms, $operator = 'IN' ) {
+		$conditions = array();
+		foreach ( $terms as $taxonomy => $terms ) {
+			if ( is_taxonomy_viewable( $taxonomy ) && ! empty( $terms ) ) {
```

Comment:
super nit, but I'd place the cheapest evaluation first:


```suggestion
			if ( ! empty( $terms ) && is_taxonomy_viewable( $taxonomy )  ) {
```